index.ts

```diff
@@ -0,0 +1,552 @@
+  megaEvolves?: SpeciesT;
+  megaStone?: SpeciesT;
+  naturalGift?: {basePower: number, type: TypeT};
+  onDrive?: TypeT;
+  onMemory?: TypeT;
+  onPlate?: TypeT;
+  status?: StatusName;
+  weather?: WeatherName;
+  zMove?: {
+    type?: TypeT;
+  } | {
+    move: MoveT;
+    from: MoveT;
+    user: SpeciesT[];
+  };
+}
+
+interface JSONItem extends JSONData && CommonItem<ID, ID, ID> {}
+interface Item extends Data & CommonItem<Species, Type, Move> {
+  kind: 'Item';
+
+  toJSON(): JSONItem;
+}
+
+class Items {
+  getByID(id: ID): Item | undefined;
+  next(): {value: Item, done: boolean};
+}
+
+// moves.ts
+
+// TODO: consider moving some top-level booleans from Move into MoveFlags
+interface MoveFlags {
+  authentic?: boolean; // Ignores a target's substitute.
+  bite?: boolean; // Power is multiplied by 1.5 when used by a Pokemon with the Ability Strong Jaw.
+  bullet?: boolean; // Has no effect on Pokemon with the Ability Bulletproof.
+  charge?: boolean; // The user is unable to make a move between turns.
+  contact?: boolean; // Makes contact.
+  dance?: boolean; // When used by a Pokemon, other Pokemon with the Ability Dancer can attempt to execute the same move.
+  defrost?: boolean; // Thaws the user if executed successfully while the user is frozen.
+  distance?: boolean; // Can target a Pokemon positioned anywhere in a Triple Battle.
+  gravity?: boolean; // Prevented from being executed or selected during Gravity's effect.
+  heal?: boolean; // Prevented from being executed or selected during Heal Block's effect.
+  mirror?: boolean; // Can be copied by Mirror Move.
+  mystery?: boolean; // Unknown effect.
+  nonsky?: boolean; // Prevented from being executed or selected in a Sky Battle.
+  powder?: boolean; // Has no effect on Pokemon which are Grass-type, have the Ability Overcoat, or hold Safety Goggles.
+  protect?: boolean; // Blocked by Detect, Protect, Spiky Shield, and if not a Status move, King's Shield.
+  pulse?: boolean; // Power is multiplied by 1.5 when used by a Pokemon with the Ability Mega Launcher.
+  punch?: boolean; // Power is multiplied by 1.2 when used by a Pokemon with the Ability Iron Fist.
+  recharge?: boolean; // If this move is successful, the user must recharge on the following turn and cannot make a move.
+  reflectable?: boolean; // Bounced back to the original user by Magic Coat or the Ability Magic Bounce.
+  snatch?: boolean; // Can be stolen from the original user and instead used by another Pokemon using Snatch.
+  sound?: boolean; // Has no effect on Pokemon with the Ability Soundproof.
+}
+
+type MoveCategory: 'Physical' | 'Special' | 'Status';
+
+type MoveTarget =
+  // single-target
+  'normal'|'any'|'adjacentAlly'|'adjacentFoe'|'adjacentAllyOrSelf'|
+  // single-target, automatic
+  'self'|'randomNormal'|
+  // spread
+  'allAdjacent'|'allAdjacentFoes'|
+  // side and field
+  'allySide'|'foeSide'|'all';
+
+interface SelfEffect {
+  volatileStatus?: ID;
+  boosts?: Partial<BoostsTable>;
```

Comment:
Can `BoostsTable` be moved up? I like to encounter definitions before their references, obviously this can't always be done, but I think should be something we should strive for

Reply:
These are meant to be in separate files - `BoostsTable` lives in `stats.ts` and would be `import`-ed